File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    unordered_map<string, int> hash;    
    for (int i=0; i<participant.size(); i++) {
        hash[participant[i]]++;
    }
    
    for (int i=0; i<completion.size(); i++) {
        hash[completion[i]]--;
    }
    for (auto i:hash) {
        if (i.second==1) answer = i.first;
    }
    return answer;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
import collections

def solution(participant, completion):
    answer = ''
#     dict = {}
#     for name in participant:
#         if not dict.get(name, 0):
#             dict[name] = 1
#         else:
#             dict[name]+=1
    
#     for name in completion:
#         dict[name]-=1
    
#     for key, value in dict.items():
#         if value:
#             return key
    
    answer = collections.Counter(participant) - collections.Counter(completion)
    return list(answer.keys())[0]
    
